core/src/main/java/org/elasticsearch/common/joda/DateMathParser.java

```diff
@@ -63,13 +63,10 @@ public long parse(String text, Callable<Long> now, boolean roundUp, DateTimeZone
         } else {
             int index = text.indexOf("||");
             if (index == -1) {
-                return parseDateTime(text, timeZone);
+                return parseDateTime(text, timeZone, roundUp);
             }
-            time = parseDateTime(text.substring(0, index), timeZone);
+            time = parseDateTime(text.substring(0, index), timeZone, false);
```

Comment:
It is not ignored. What happens is that the date is rounded explicitly using the `||` notation. Eg. `2010-03-19 09:45:02||/d` which tells Elasticsearch to parse the date and round it with a daily interval. So first we parse the date without taking care of rounding (the line that you are commenting) and then we are applying the rounding 2 lines below in the `parseMath` method.